tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -136,13 +139,35 @@ public static void Initialize()
                 Log.Error(ex, ex.Message);
             }
 
-            InitializeNoNativeParts();
+            InitializeNoNativeParts(out Tracer tracer);
+
+            if (tracer is null)
+            {
+                Log.Debug("Skipping TraceMethods initialization because InitializeNoNativeParts did not return a Tracer instance");
+            }
+            else
+            {
+                try
+                {
+                    Log.Debug("Running TraceMethods initialization because InitializeNoNativeParts returned a Tracer instance");
+                    var traceMethodsConfiguration = tracer.Settings.TraceMethods;
+                    if (!string.IsNullOrEmpty(traceMethodsConfiguration))
+                    {
+                        NativeMethods.InitializeTraceMethods(initializeProfilerPayloadDefinitionId, traceMethodsConfiguration);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, ex.Message);
+                }
+            }
 
             Log.Debug("Initialization finished.");
         }
 
-        internal static void InitializeNoNativeParts()
+        internal static void InitializeNoNativeParts(out Tracer tracer)
```

Comment:
nit - I would have returned a boolean saying that the tracer has ben initialized and called Tracer.Instance directly.

Reply:
Thanks for the feedback. I decided to revert my changes to this method so it's still `void`, and after the call I just check if `Tracer.Instance` is set or not. What do you think? You can view these changes in ac173c08b1b9c3c39876e5b55ae9edacc90bbf39

Reply:
Indeed, it's even better